Add regular expression to SSG glossary (#255)
redhat-documentation/supplementary-style-guide: supplementary_style_guide/glossary_terms_conventions/general_conventions/r.adoc

*See also*: xref:zone-group[zone group]

[discrete]
[[regex]]
==== image:images/no.png[no] regex (noun)

> *Description*: _Regex_ is an abbreviation for "regular expression". Do not use.

Reviewer: I know it's in the "see also" section, but could you incorporate guidance to use "regular expression"? Maybe something like:

```suggestion
*Description*: _Regex_ is an abbreviation for "regular expression". Do not use; use "regular expression" instead.
```

or maybe even just:

```suggestion
*Description*: Do not use "regex" as a replacement for "regular expression".
```

*Use it*: no

*Incorrect forms*:

*See also*: xref:regular-expression[regular expression]

[discrete]
[[regular-expression]]

> ==== image:images/yes.png[no] regular expression (noun)

Reviewer: Can you fix this alt text to "yes"?

```suggestion
==== image:images/yes.png[yes] regular expression (noun)
```

> *Description*: _Regular expression_ is a string of text specifying a pattern that a computer can use when searching a body of text to match characters or groups of characters. 

Reviewer: Maybe start the sentence with "A" for it to read more smoothly?

```suggestion
*Description*: A _regular expression_ is a string of text specifying a pattern that a computer can use when searching a body of text to match characters or groups of characters. 
```

> *Description*: A _regular expression_ is a string of text specifying a pattern that a computer can use when searching a body of text to match characters or groups of characters. 

Reviewer 2: ```suggestion
*Description*: A _regular expression_ is a string of letters, numbers, and symbols that defines a pattern. When searching a body of text, software can use this pattern to match characters or groups of characters and return results.
```
- To my ears, this sentence seems a bit too long. Maybe consider shortening it by breaking it into two sentences. My edit shows one possibility.
- Consider saying `string of letters, numbers, and symbols` rather than `text` to define the term more specifically. Your call...
- Consider moving the `when searching` clause in front of `a computer` to simplify the syntax and to avoid possibly unclear modifiers in the dependent clauses.
- Consider specifying `computer program` or `software` rather than just `computer` because computer hardware by itself can't really perform searches without software.
- Consider adding `and return results` to the end because software not only matches the defined pattern but returns the results for further processing.

*Use it*: yes

*Incorrect forms*: regex

> *See also*: 

Reviewer: Can you also link to the regex entry?
